os/kernel/debug/sysdbg.c

```diff
@@ -416,60 +326,86 @@ static void sysdbg_print(void)
 
 static ssize_t sysdbg_write(FAR struct file *filep, FAR const char *buffer, size_t buflen)
 {
-	char arg1[32] = { '\0' };
-	char arg2[32] = { '\0' };
+	char option[32] = { '\0' };
+	char count[32] = { '\0' };
+
+	sscanf(buffer, "%s %s", option, count);
 
-	sscanf(buffer, "%s %s", arg1, arg2);
+	if (option[0] == '\0') {
+		dbg("Invalid argrument.\n");
+		show_usage();
+		return -1;
+	}
+
+	if (strncmp("help", option, 5) == 0) {
+		show_usage();
+		return 0;
+	}
 
-	if (strncmp("enable_monitor", arg1, 14) == 0) {
+	switch (atoi(option)) {
+	case SYSDBG_ENABLE_MONITOR:		/*Enable monitor*/
 		sysdbg_monitor_enable();
-	} else if (strncmp("disable_monitor", arg1, 15) == 0) {
+		break;
+	case SYSDBG_DISABLE_MONITOR:		/*Disable monitor*/
 		sysdbg_monitor_disable();
-	} else if (strncmp("read", arg1, 4) == 0) {
+		break;
+	case SYSDBG_PRINT:			/*Read /dev/sysdbg to view debug information*/
 		sysdbg_print();
-	} else if (strncmp(arg1, "max_task_count", 14) == 0) {
+		break;
+	case SYSDBG_UPDATE_MAX_TASK_CNT:	/*Update max task count*/
 #ifdef CONFIG_TASK_SCHED_HISTORY
-		update_maxtask_count(atoi(arg2));
+		if (count[0] == '\0') {
+			dbg("Enter the count in power of 2 and > 0.\n");
+			dbg("For help type : sysdbg help\n");
+			return -1;
+		}
+		update_maxtask_count(atoi(count));
 #else
 		dbg("To use this feature, kindly enable CONFIG_TASK_SCHED_HISTORY\n");
 #endif
-	} else if (strncmp(arg1, "max_irq_count", 13) == 0) {
+		break;
+	case SYSDBG_UPDATE_MAX_IRQ_CNT:		/*Update maxirq_count*/
 #ifdef CONFIG_IRQ_SCHED_HISTORY
-		update_maxirq_count(atoi(arg2));
+		if (count[0] == '\0') {
+			dbg("Enter the count in power of 2 and > 0.\n");
+			dbg("For help type : sysdbg help\n");
+			return -1;
```

Comment:
Can you please use standard error values or add #define instead of returning "-1"?